libshaderc/include/shaderc.hpp

```diff
@@ -219,6 +219,20 @@ class CompileOptions {
     shaderc_compile_options_set_warnings_as_errors(options_);
   }
 
+  // Sets the weak shader kind to be used for following compilation. The
+  // shader kind set here has lowest priority compared with the
+  // forced_shader_kind passed through shaderc_compile_into_spv()
+  // and #pragma annotation in source text. The compiler will check
+  // the forced_shader_kind first. If the forced_shader_kind is
+  // shaderc_glsl_none_shader, the compiler will check the #pragma
+  // annotation in the source text. If the #pragma annotation is not set,
+  // the compiler will use the shader kind set by this function to compile
+  // the source text.
+  void SetWeakShaderKind(shaderc_shader_kind kind) {
```

Comment:
We discussed this in person, and decided to expand the shaderc_shader_kind so it has:
 - one section of values with plain names than mean "force vertex shader", for example
 - one value of "get shader kind from pragma annotation"
 - another section of values such as "default to vertex", for example.

That way you have *one* place/argument to specify all the interacting priorities.  (This avoids weird interactions like what does override_pragmas true mean with shader kind none?  Error?)
Also, the simplest-named ones are presented first and have the force semantics, so they are the easiest to use.